editoast/src/views/train_schedule/simulation_report.rs

```diff
@@ -180,6 +198,62 @@ async fn project_simulation_results(
     })
 }
 
+async fn add_stops_additional_information(
+    stops: Vec<ResultStops>,
+    infra_id: i64,
+    path_waypoints: &[PathWaypoint],
+    db_pool: Data<DbPool>,
+) -> error::Result<Vec<FullResultStops>> {
+    let mut conn = db_pool.get()?;
+    let track_ids: Vec<String> = path_waypoints.iter().map(|pw| pw.track.clone()).collect();
+    let track_sections: Vec<TrackSectionModel> = sql_query(include_str!("sql/track_sections.sql"))
+        .bind::<BigInt, _>(infra_id)
+        .bind::<SqlArray<Text>, _>(&track_ids)
+        .load(&mut conn)?;
+    let track_sections_map: HashMap<String, TrackSectionModel> = HashMap::from_iter(
+        track_sections
+            .iter()
+            .map(|ts| (ts.obj_id.clone(), ts.clone())),
+    );
+    let stops = stops
+        .iter()
+        .zip(path_waypoints.iter())
+        .map(|(s, pw)| {
+            match &track_sections_map
+                .get(&pw.track)
+                .unwrap()
+                .data
+                .extensions
+                .sncf
+            {
+                Some(ext) => FullResultStops {
+                    time: s.time,
+                    position: s.position,
+                    duration: s.duration,
+                    id: pw.id.clone(),
+                    name: pw.name.clone(),
+                    line_code: Some(ext.line_code),
+                    track_number: Some(ext.track_number),
+                    line_name: Some(ext.line_name.to_string()),
+                    track_name: Some(ext.track_name.to_string()),
+                },
+                None => FullResultStops {
+                    time: s.time,
+                    position: s.position,
+                    duration: s.duration,
+                    id: None,
+                    name: None,
+                    line_code: None,
+                    track_number: None,
+                    line_name: None,
+                    track_name: None,
+                },
+            }
+        })
+        .collect();
+    Ok(stops)
+}
+
 fn project_speeds(mut speeds: Vec<ResultSpeed>, departure_time: f64) -> Vec<ResultSpeed> {
     for speed in speeds.iter_mut() {
         speed.time += departure_time;
```

Comment:
Same here. Whole new package of information is added, but not tested.

Reply:
All these changes are traversed by the test made in stdcm